src/backend/app/projects/image_processing.py

```diff
@@ -0,0 +1,155 @@
+import os
+import uuid
+import tempfile
+import shutil
+from pyodm import Node
+from app.s3 import get_file_from_bucket, list_objects_from_bucket, add_file_to_bucket
+from loguru import logger as log
+from concurrent.futures import ThreadPoolExecutor
+
+
+class DroneImageProcessor:
+    def __init__(
+        self,
+        node_odm_url: str,
+        project_id: uuid.UUID,
+        task_id: uuid.UUID,
+    ):
+        """
+        Initializes the connection to the ODM node.
+        """
+        # self.node = Node(node_odm_host, node_odm_port)
+        self.node = Node.from_url(node_odm_url)
+        self.project_id = project_id
+        self.task_id = task_id
+
+    def options_list_to_dict(self, options=[]):
+        """
+        Converts options formatted as a list ([{'name': optionName, 'value': optionValue}, ...])
+        to a dictionary {optionName: optionValue, ...}
+        """
+        opts = {}
+        if options is not None:
+            for o in options:
+                opts[o["name"]] = o["value"]
+        return opts
+
+    def download_object(self, bucket_name: str, obj, images_folder: str):
+        if obj.object_name.endswith((".jpg", ".jpeg", ".JPG", ".png", ".PNG")):
+            # log.info(f"Downloading image from s3 {obj.object_name}")
+            local_path = f"{images_folder}/{os.path.basename(obj.object_name)}"
+            os.makedirs(os.path.dirname(local_path), exist_ok=True)
+            get_file_from_bucket(bucket_name, obj.object_name, local_path)
+
+    def download_images_from_minio(self, bucket_name, local_dir):
+        """
+        Downloads images from MinIO under the specified path.
+
+        :param bucket_name: Name of the MinIO bucket.
+        :param project_id: The project UUID.
+        :param task_id: The task UUID.
+        :param local_dir: Local directory to save the images.
+        :return: List of local image file paths.
+        """
+        prefix = f"projects/{self.project_id}/{self.task_id}"
+
+        objects = list_objects_from_bucket(bucket_name, prefix)
+
+        # Process images concurrently
+        with ThreadPoolExecutor() as executor:
```

Comment:
I would recommend asyncio over threading for this type of downloading onto a filesystem.

`asyncio.gather` would work nicely and be less blocking (using a single thread via coroutines).

Example from FMTM:

https://github.com/hotosm/fmtm/blob/35e7240cc78d62c288f0ea60aef0eb7bde8beaed/src/backend/app/central/central_crud.py#L560-L570

But not this would mean you need to update functions to `async` and use `await`